### DISCUSSION TITLE
Hacktoberfest repository topic

### DISCUSSION THREAD

**Discussion by FiniteReality** [dotnet/runtime#43005] [General]
According to the DigitalOcean, they have changed the rules for Hacktoberfest this year (https://hacktoberfest.digitalocean.com/hacktoberfest-update) - this means that only repositories with the `hacktoberfest` topic count towards the four PRs people may open during October.

I think it would be nice to add to the pool of repositories which are valid for hacktoberfest PRs, so it would be nice if we could add the hacktoberfest topic here, and maybe other .NET repositories. (E.g. roslyn, docs, etc.)

However, due to the small number of repositories with this topic, it may encourage more spam PRs, which is why I have opened this as a discussion rather than an issue.

**Comment by tannergooding** [comment]
@terrajobst, @danmosemsft could one of you update the repo to include the topic (the `hacktoberfest-update` link has a short gif showing how)?

We were already planning on participating and while other repos had seen some spam, we hadn't. I believe the new rules (which require PRs to be merged, approved, or marked `hacktoberfest-accepted`) will also help cut back on any spam that might otherwise appear.

**Reply by danmoseley** [reply]
I'm not an admin - @terrajobst would have o do it.

**Comment by terrajobst** [comment]
I’d rather they inverted it: look for merged PRs submitted by first timers or merged PRs marked with some label like “well done”.

As it stands, it’s a factory for low quality PRs.

**Reply by tannergooding** [reply]
Could you elaborate on how it is a factory for low quality PRs?

The new rules are:
```
Submitted in a repo with the hacktoberfest topic AND
during the month of October AND (
  The PR is merged OR
  The PR is labelled as hacktoberfest-accepted by a maintainer OR
  The PR has been approved
)
```

The requirement that the repo be participating and that it be merged should prevent most low effort PRs, as they wouldn't be merged or accepted.

**Reply by FiniteReality** [reply]
I agree that there are a few issues with the changes they've made, and people have raised their voices about it already.

@tannergooding I believe the main issue is that only a few repositories have the `hacktoberfest` topic, and a large percentage of them are low-effort PR farms. By adding the topic, it's possible that you're putting your repository up in the face of a large portion of people who end up creating PRs which simply change whitespace or similar.

**Reply by danmoseley** [reply]
This all seems hypothetical - why don't we just do it and see. We can remove the topic again if necessary I assume. It doesn't take much effort to close a spam PR.

**Reply by terrajobst** [reply]
> Could you elaborate on how it is a factory for low quality PRs?

I see it all over social media. There are numerous PRs submitted to many repos. They changed the rules after it started this year because of it. The new rules seem good enough, but we'll see. I have the topic applied.
